src/App.js

```diff
@@ -35,14 +36,26 @@ class App extends React.Component {
     this.state = {
       todoItems,
       //Q: why is this kept here? it isn't a slice of state we're tracking, is it? Does this enable the rerendering when it changes???
-      toggleCompleted: () => {},
-      removeCompleted: () => {},
+      //Yetti says These are definitely weird here & I didn't need them.
+      // toggleCompleted: () => {},
+      // removeCompleted: () => {},
+      searchTerm: "",
     };
   }
 
+  // Search attempt
+  editSearchTerm = (value) => {
+    this.setState({searchTerm: value})
+  }
+  dynamicSearch = () => {
+    const { todoItems, searchTerm } = this.state;
+    return todoItems.filter(todo => todo.task.toLowerCase().includes(searchTerm.toLowerCase()))
+  }
+
   toggleCompleted = (clickedItemId) => {
+    const { todoItems } = this.state;
```

Comment:
Lots of deconstructuring helped me understand so much better, plus make the code far more readable. Christopher highly, highly, highly emphasises deconstructuring EVERYTHING